### PR TITLE
fix dns modifications inside of the kube-proxy for windows userspace

### LINKED ISSUE #102846: windows userspace proxy has an extra dependency w/ unwanted behaviour
#### What happened:

Windows userspace proxy has some unnecessary docker DNS mangling stuff in it, discovered by @aojea  a long time ago, https://patch-diff.githubusercontent.com/raw/kubernetes/kubernetes/pull/97812.patch 

#### What you expected to happen 

we wouldnt have extra dead code in windows userspace proxy, because after all userspace proxying is the future of Kubernetes (sarcasm).  but either way we need to clean this up :)

### PR DESCRIPTION
**Pull request by jayunit100**
Fixes #102846 

Thanks again @aojea for this ! im stealing your original PR and removing the bazel bits

### CODE REVIEW

**Comment by aojea** [comment]
/assign @dims @liggitt 
/kind cleanup
one dependency less

**Comment by dims** [comment]
@jayunit100 looks like you need to cleanup the commits to pass the CLA check

**Comment by liggitt** [comment]
happy to see the vendoring go away, will defer to windows/network folks for the functional changes

**Comment by aojea** [comment]
are you doing hack/update-vendor.sh?
it seems this is pulling new dependencies and not fully removing the miekg dns one

File: go.mod [hunk 1]
```diff
@@ -61,7 +61,6 @@ require (
 	github.com/libopenstorage/openstorage v1.0.0
 	github.com/lithammer/dedent v1.1.0
 	github.com/lpabon/godbc v0.1.1 // indirect
-	github.com/miekg/dns v1.1.35
```
Comment (liggitt) [inline go.mod:64]: follow the instructions at the top of the file for updating:

```
// Ensure you've carefully read
// https://git.k8s.io/community/contributors/devel/sig-architecture/vendor.md
// Run hack/pin-dependency.sh to change pinned dependency versions.
// Run hack/update-vendor.sh to update go.mod files and the vendor directory.
```

run `hack/lint-dependencies.sh` locally to make sure everything is up to date before committing/pushing

Comment (jayunit100) [reply go.mod:64]: thanks ! ok, i just redid the entire vendor of deps and that fixed it

**Comment by jayunit100** [comment]
@aojea can you review the kube-proxy parts ? would like to try to merge this wk since its a no-op for most people other then me :) :)

**Comment by jayunit100** [comment]
@aojea @dims all set on this ?

**Comment by liggitt** [comment]
is the functionality this is removing exercised by the windows presubmits?

**Comment by dims** [comment]
@jayunit100 let's get reviews from sig-windows folks and see if you can trigger a presubmit that exercises this code in a windows environment as well.

/release-note-none

**Comment by jayunit100** [comment]
this is windows userspace. 
Im the only person in the entire world that runs the kube-proxy in windows userspace, and
ive tested it :)

The code for testing this is here https://github.com/FriedrichWilken/KubernetesOnWindows/,
and the way to test it is to vagrant up the VMs and swap this out with 
https://storage.googleapis.com/jayunit100/kube-proxy-windows-dns-fix.exe 

@timothysc will get a kick out of reading this . but its all totally 100% true.  We still need vagrant in 2021 :)

**Comment by weiwenli97** [comment]
I also verified kube-proxy-windows-dns-fix.exe on the windows workload including 3 Linux control plane and 3 Windows workers. Here are the results https://kscom.svc.eng.vmware.com/job/self-service/job/windows-sandbox-3/286/consoleFull, including the windows nodeport tests, so it looks like userspace windows proxy patch is valid.

> 17:51:28  • [SLOW TEST:100.634 seconds]
> 17:51:28  [sig-windows] Services
> 17:51:28  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/windows/framework.go:27
> 17:51:28    should be able to create a functioning NodePort service for Windows
> 17:51:28    /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/windows/service.go:42
> 17:51:28  ------------------------------
> 17:51:28  {"msg":"PASSED [sig-windows] Services should be able to create a functioning NodePort service for Windows","total":10,"completed":5,"skipped":2204,"failed":0}

> 
> 17:52:48  Ran 9 of 6398 Specs in 900.340 seconds
> 17:52:48  SUCCESS! -- 9 Passed | 0 Failed | 0 Pending | 6389 Skipped
> 17:52:48  PASS

**Comment by jayunit100** [comment]
thanks for the extra testing @weiwenli97 ... FYI Wenlis tests were on a variant of TKG, w/ k8s version 1.21.  Mine were on 1.20....

**Pull request merged**